### PR TITLE
[Feature] Per Command Cost.

### PR DESCRIPTION
**Pull request by d0cd**
This PR assigns a cost to each `Command` variant.

Depends on #1636

### CODE REVIEW

**Review by howardwu — APPROVED** [review]
LGTM

**Comment by howardwu** [comment]
Should we consider moving the cost numbers to somewhere more visible?

File: synthesizer/src/program/finalize/mod.rs [hunk 1]
```diff
@@ -74,9 +74,91 @@ impl<N: Network> Finalize<N> {
     }
 
     /// Returns the minimum fee, in microcredits, required to run the finalize.
-    // TODO (d0cd): Introduce a cost table for each instruction and use that to calculate the fee.
     pub fn fee_in_microcredits(&self) -> u64 {
```
Comment (howardwu) [inline synthesizer/src/program/finalize/mod.rs]: Could we rename this to `cost_in_microcredits`?

**Comment by d0cd** [comment]
We could definitely move it around.
Perhaps into a file called `cost.rs` in the `finalize` module? Do you have any preferences?
Alternatively, if its referenced well enough in documentation, we can leave it be.

File: synthesizer/src/program/finalize/mod.rs [hunk 2]
```diff
@@ -73,10 +73,92 @@ impl<N: Network> Finalize<N> {
...
+        // Defines the cost of each command.
+        let cost = |command: &Command<N>| match command {
+            Command::Instruction(Instruction::Abs(_)) => 2_000,
+            Command::Instruction(Instruction::AbsWrapped(_)) => 2_000,
+            Command::Instruction(Instruction::Add(_)) => 2_000,
+            Command::Instruction(Instruction::AddWrapped(_)) => 2_000,
+            Command::Instruction(Instruction::And(_)) => 2_000,
+            Command::Instruction(Instruction::AssertEq(_)) => 2_000,
+            Command::Instruction(Instruction::AssertNeq(_)) => 2_000,
+            Command::Instruction(Instruction::Call(_)) => 1_000_000_000_000, // Note: `call` is not allowed in a finalize.
```
Comment (howardwu) [inline synthesizer/src/program/finalize/mod.rs]: I would just change this method's return type to `Result<u64>` and write a `bail!`

**Review by howardwu — APPROVED** [review]
LGTM

**Pull request merged**